### PR TITLE
Keep best CP pokemon

### PR DESCRIPTION
**Pull request by koper89**
Short Description: 
Couldn't test it cause dev branch cannot find me any pokemons, but hopes it'll work.

### CODE REVIEW

**Comment by koper89** [comment]
if not self._check_if_higher_cp_pokemon(pokemon_name, cp):
 +		return True

in here should be False

**Comment by Calcyfer** [comment]
The logic seems correct, I'll test it in my patched dev branch

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -324,8 +356,12 @@ def should_release_pokemon(self, pokemon_name, cp, iv, response_dict):
 
         if release_config.get('always_release', False):
             return True
-
-        release_cp = release_config.get('release_below_cp', 0)
+	
+	if release_config.get('keep_best_cp', True):
```
Comment (Calcyfer) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Maybe we would want the default value to be False, so nothing changes for users who pull but don't update their release_config.json file?

**Comment by Calcyfer** [comment]
Currently getting this error : 

`pokemon_catch_worker.py", line 338, in _get_catch_config_for
    catch_config = self.config.catch.get(pokemon)
`
`AttributeError: 'Namespace' object has no attribute 'catch'
`

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (koper89) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Yep but in dev branch there's no more realeae congig as separate file

**Comment by koper89** [comment]
Doesn't seem like issue with my addition

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (Calcyfer) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: You're right. This just caused quite a few errors :)

**Comment by Calcyfer** [comment]
It's not, that's why I removed the comment - damn all these commits.

Works great on dev btw, after fixing the issues with the config created by other commits.

**Comment by koper89** [comment]
Great to hear.

**Comment by Nihisil** [comment]
👍  

I hope you have enough patience to merge all dev changes into your branch and that you will not give up with this pull-request.

**Comment by slimjim2234** [comment]
LGTM!
:+1:

**Comment by namelesswc** [comment]
Suggestion: It should keep lower cp pokemon in case it has very good iv otherwise I'm gonna loose all my perfekt pokemons.

**Comment by koper89** [comment]
https://github.com/PokemonGoF/PokemonGo-Bot/pull/953
Moved to this pull.

**Pull request closed without merging**